test/e2e/virtual/apiexport/authorizer_test.go

```diff
@@ -346,3 +354,175 @@ func TestAPIExportAuthorizers(t *testing.T) {
+	require.NoError(t, err)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(serviceProviderKcpClient.Cluster(serviceWorkspace).Discovery()))
+	err = helpers.CreateResourceFromFS(ctx, serviceDynamicClusterClient.Cluster(serviceWorkspace), mapper, nil, "apiresourceschema_cowboys.yaml", testFiles)
+	require.NoError(t, err)
+
+	t.Logf("Get the root scheduling APIExport's identity hash")
+	schedulingAPIExport, err := kcpClient.Cluster(tenancyv1alpha1.RootCluster).ApisV1alpha1().APIExports().Get(ctx, "scheduling.kcp.dev", metav1.GetOptions{})
+	require.NoError(t, err)
+	require.True(t, conditions.IsTrue(schedulingAPIExport, apisv1alpha1.APIExportIdentityValid))
+	identityHash := schedulingAPIExport.Status.IdentityHash
+	require.NotNil(t, identityHash)
+
+	t.Logf("Create an APIExport for APIResourceSchema in service provider %q", serviceWorkspace)
+	apiExport := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "today-cowboys",
+		},
+		Spec: apisv1alpha1.APIExportSpec{
+			LatestResourceSchemas: []string{"today.cowboys.wildwest.dev"},
+			PermissionClaims: []apisv1alpha1.PermissionClaim{
+				{
+					GroupResource: apisv1alpha1.GroupResource{Group: scheduling.GroupName, Resource: "placements"},
+					IdentityHash:  identityHash,
+					All:           true,
+				},
+			},
+		},
+	}
+	apiExport, err = serviceKcpClient.Cluster(serviceWorkspace).ApisV1alpha1().APIExports().Create(ctx, apiExport, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Grant user to be able to bind service API export from workspace %q", serviceWorkspace)
+	cr, crb := createClusterRoleAndBindings(
+		consumerUser,
+		consumerUser, "User",
+		[]string{"bind"},
+		apis.GroupName, "apiexports", apiExport.Name,
+	)
+	_, err = kubeClient.Cluster(serviceWorkspace).RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = kubeClient.Cluster(serviceWorkspace).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Create an APIBinding in consumer workspace %q that points to the service APIExport from %q", userWorkspace, serviceWorkspace)
+	apiBinding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cowboys",
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					Path:       serviceWorkspace.String(),
+					ExportName: apiExport.Name,
+				},
+			},
+			PermissionClaims: []apisv1alpha1.AcceptablePermissionClaim{
+				{
+					PermissionClaim: apisv1alpha1.PermissionClaim{
+						GroupResource: apisv1alpha1.GroupResource{Group: scheduling.GroupName, Resource: "placements"},
+						IdentityHash:  identityHash,
+						All:           true,
+					},
+					State: apisv1alpha1.ClaimAccepted,
+				},
+			},
+		},
+	}
+	err = retry.OnError(retry.DefaultBackoff, errors.IsForbidden, func() error {
```

Comment:
Out of curiosity, why is it being forbidden?

Reply:
It happens the ClusterRole and ClusterRoleBinding created few lines above do not apply synchronously, so the creation of the APIBinding can fail until this applies. That is actually the reason why I favoured `retry.OnError` over `framework.Eventually`, because it's only useful to retry on 403, and it can fail-fast otherwise.

Reply:
@stevekuznetsov wdyt?

Reply:
Next commit changes to use `framework.Eventually` - but to the point I'd say it's highly unlikely we get an error we can fail fast on and the benefit from the eventually bit is good. We can try to improve `framework.Eventually` for a short-circuit exit if we want